solutions/usaco-1179.mdx

```diff
@@ -0,0 +1,90 @@
+---
+id: usaco-1179
+source: USACO Gold 2022 January
+title: Herdle
+author: Juheon Rhee
+---
+
+[Official Editorial (Java)](http://www.usaco.org/current/data/sol_prob1_bronze_jan22.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main(){
+
+	char correct[9], guess[9];
+
+	for(int i = 0; i < 9; i++){
+		cin >> correct[i];
+	}
+	
+	for(int i = 0; i < 9; i++){
+		cin >> guess[i];
+	}
+
+	int green = 0, yellow = 0;
+
+	for(int i = 0; i < 9; i++){
+		if(correct[i] == guess[i]){
+			correct[i] = guess[i] = '0';
+			green ++;
+		}
+	}
+    
+	for(int i = 0; i < 9; i++){
+		if(correct[i] == '0'){
+			continue;
+		}
+	
+		for(int j = 0; j < 9; j++){
+			if(guess[j] == '0'){
+				continue;
+			}
+			if(correct[i] == guess[j]){
+				correct[i] = guess[j] = '0';
+				yellow ++;
+				break;
+			}
+		}
+	}
+				     
+	cout << green << '\n' << yellow;
+	return 0;
+}
+```
+
+</CPPSection>
+
+<PySection>
+
+```python
+def getInput():
+	k = input() + input() + input()
+	return [i for i in k]
+
+correct, guess = getInput(), getInput()
+green, yellow = 0, 0
+
+for i in range(9):
+	if correct[i] == guess[i]:
+		correct[i], guess[i] = "visited", "visited"
```

Comment:
perhaps keep an auxillary array, i don't think this is the best way to do this really

Reply:
okay i just like my code short.. but i understand that it might not be the best impl

Reply:
bump